be1-go/hub/organizer.go

```diff
@@ -484,3 +502,161 @@ func (c *laoChannel) processRollCallObject(data message.Data) error {
 
 	return nil
 }
+func (c *laoChannel) processElectionObject(msg message.Message) error {
+	action := message.ElectionAction(msg.Data.GetAction())
+
+	if action != message.ElectionSetupAction {
+		return &message.Error{
+			Code:        -1,
+			Description: fmt.Sprintf("invalid action: %s", action),
+		}
+	}
+	err := c.createElection(msg)
+	if err != nil {
+		return xerrors.Errorf("failed to setup the election", err)
+	}
+
+	return nil
+}
+
+func (c *laoChannel) createElection(msg message.Message) error {
+	o := c.hub
+
+	o.Lock()
+	defer o.Unlock()
+
+	// Check the data
+	data, ok := msg.Data.(*message.ElectionSetupData)
+
+	if !ok {
+		return &message.Error{
+			Code:        -4,
+			Description: "failed to cast data to SetupElectionData",
+		}
+	}
+
+	// Check if the Lao ID of the message corresponds to the channel ID
+	encodedLaoID := base64.StdEncoding.EncodeToString(data.LaoID)
+	channelID := c.channelID[6:]
+	if channelID != encodedLaoID {
+		return &message.Error{
+			Code:        -4,
+			Description: fmt.Sprintf("Lao ID of the message (Lao: %s) is different from the channelID (channel: %s)", encodedLaoID, channelID),
+		}
+	}
+
+	// Compute the new election channel id
+	encodedElectionID := base64.StdEncoding.EncodeToString(data.ID)
+	encodedID := encodedLaoID + "/" + encodedElectionID
+
+	// Create the new election channel
+	electionCh := electionChannel{
+		createBaseChannel(o, "/root/"+encodedID),
+		data.Questions[0].VotingMethod,//TODO : check if this is what was meanth by method ot is it rather pluralityy and stuff??,
+		//Also can it happen that there is a valid setup without questions?
+		data.StartTime,
+		data.EndTime,
+		false,
+		getAllQuestionsForElectionChannel(data.Questions),
+	}
+
+	// Add the SetupElection message to the new election channel
+	messageID := base64.StdEncoding.EncodeToString(msg.MessageID)
+	electionCh.inbox[messageID] = msg
+	electionCh.inbox["0"] = msg//have this message saved at a place by
```

Comment:
I don't understand why we need to put the message here. Can you explain?

Reply:
This was before you added the useful fields in the `electionChannel` so indeed this is not needed anymore